lazy_dataset/core.py

```diff
@@ -646,7 +646,9 @@ def filter(self, filter_fn: callable, lazy: bool = True) -> 'Dataset':
                     'You can only use lazy=False if the incoming dataset is '
                     'indexable.'
                 )
-            return self[[i for i, e in enumerate(self) if filter_fn(e)]]
+            idx = [i for i, e in enumerate(self) if filter_fn(e)]
+            LOG.info(f'Filtered {len(self)-len(idx)} from {len(self)} examples.')
```

Comment:
Do we want to print this message when nothing was filtered? The message in `CatchExceptionDataset` is only printed when an exception was caught.

Reply:
No, you are right. Fixed now.